Add Description field to Project

kkangs0226 opened the pull request:
Add Description and DescriptionTest class, as well as make relevant changes in other files to add Description as a field under Project. 

closes #17, closes #21

kkangs0226 committed:
Add Description field to Project

Description is another field that can be added to Project, which
has not been implemented yet.

Adding description would allow users to add more information
to their projects.

Let's
* add Description class to project
* add DescriptionTest to project
* make changes to other related files to include Description class

pr4aveen commented on src/main/java/seedu/momentum/model/project/Description.java:
```diff
@@ -0,0 +1,42 @@
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represent a Project's description in teh project book.
```
Minor typo here

kkangs0226 replied:
Fixed!

pr4aveen commented on src/test/java/seedu/momentum/logic/parser/AddCommandParserTest.java:
```diff
@@ -64,34 +72,28 @@ public void parse_compulsoryFieldMissing_failure() {
+        assertParseFailure(parser, VALID_NAME_BOB + VALID_DESCRIPTION_BOB,
                 expectedMessage);
 
-        // missing phone prefix
-        //        assertParseFailure(parser, NAME_DESC_BOB + VALID_PHONE_BOB,
-        //                expectedMessage);
-
```
Is there a variation of this test that you have written?

kkangs0226 replied:
Nope, there are no other compulsory fields for Project other than name since description and tags are optional

pr4aveen commented on src/main/java/seedu/momentum/logic/parser/ArgumentMultimap.java:
```diff
@@ -33,12 +33,26 @@ public void put(Prefix prefix, String argValue) {
     }
 
+    /**
+     * getValue method for PREFIX_DESCRIPTION in parse method of AddCommandParser class
+     * Returns the last value of {@code prefix}.
+     * If the prefix does not exist or has no values, this will return an Optional.empty.
```
Not sure if this Javadoc adheres to the coding standard

kkangs0226 replied:
Fixed!

pr4aveen reviewed (approved):
LGTM 👍

khoodehui commented on src/main/java/seedu/momentum/model/project/Project.java:
```diff
@@ -18,21 +18,28 @@
     // Identity fields
     private final Name name;
 
+    //data fields
```
Perhaps there should be a whitespace after the slashes?

kkangs0226 replied:
Fixed!

boundtotheearth commented on src/main/java/seedu/momentum/logic/parser/ArgumentMultimap.java:
```diff
@@ -33,12 +33,26 @@ public void put(Prefix prefix, String argValue) {
+     * Returns the last value of {@code prefix}.
+     * If the prefix does not exist or has no values, this will return an Optional.empty.
+     */
+    public Optional<String> getAddCommandDescriptionValue(Prefix prefix) {
```
What is the purpose of this method? Is it not possible to use/change the existing getValue() method?

kkangs0226 replied:
I changed the code for the parse method of AddCommandParser class instead so the existing getValue() method can be used, thanks for the comment

boundtotheearth commented on src/main/java/seedu/momentum/model/project/Project.java:
```diff
@@ -51,7 +58,7 @@ public boolean isSameProject(Project otherProject) {
 
         return otherProject != null
-                && otherProject.getName().equals(getName());
+                && otherProject.getName().equals(getName()) && otherProject.getDescription().equals(getDescription());
```
Should description be considered an identity field for a project? I.e. should we allow projects with same name but different description?

kkangs0226 replied:
Will leave it as such as discussed, will discuss more in detail after implementation of project ID

boundtotheearth commented on src/test/data/JsonProjectBookStorageTest/invalidProjectProjectBook.json:
```diff
@@ -1,5 +1,6 @@
   "projects": [ {
-    "name": "Project with invalid name field: Ha!ns Mu@ster"
+    "name": "Project with invalid name field: Ha!ns Mu@ster",
+    "description": "description"
```
I think it would be useful to implement the Json serialization such that if description does not exist, then it is taken to be empty (""), so we don't have to add a description to all the sample data

kkangs0226 replied:
Good suggestion, but could we push this to our next iteration if it is not too urgent? I am still not very familiar with how Json works, will need some time to work on it!

khoodehui reviewed (approved):
Overall LGTM!

kkangs0226 committed:
Fix coding standard violations and remove redundant method

getAddCommandDescriptionValue method in the ArgumentMultimap class
can be removed by editing the parse method in the AddCommandParser
class instead.

Removing the method and fixing codign standards would make the
code cleaner.

Let's
* remove redundant method
* fix coding standard violations
